### ISSUE TITLE
Need An API to Determine What Level of Code The Fabric is Running

### ISSUE THREAD

**Issue by bmos299** [hyperledger-archives/fabric#1432]
I need some API (REST and/or GRPC) to determine what level of code my peer is running.

For example,

There is a network api:
GET host:port/network/peers

I think we need new api for admin that we can continue to build on.  For example, this first call could be:
GET host:port/admin/commit

This would return the code level for all peers.  This will allow us to easily determine the code level and if there is a code level mismatch.  Here is an example,
{
    "peers": [
        {
            "ID": {
                "name": "vp1"
            },
            "commit": " 2c3af309a711d3d83f6687e1722652eba5e1de81",
        },
        {
               "ID": {
                "name": "vp2"
            },
            "commit": " 2c3af309a711d3d83f6687e1722652eba5e1de81",
        }
    ]
}

**Comment by christo4ferris** [comment]
actually, IMO every call should return some indication of what version of the software processed the request in an API call... and SDK should indicate what version is expected in a request header and there should be some sort of process for a) alerting the client that it needs a more recent version of the software and b) errors if the client version is incompatible (note that both may apply).

**Comment by bmos299** [comment]
@christo4ferris absolutely.   I definitely see this being much more extensive than this issue.  It is much more of a detailed requirement.   For the short term, I am just hoping to have some ability to know what version of the fabric I'm running while design for a more eloquent solution is developed.

**Comment by chetmurthy** [comment]
@bmos299 @christo4ferris  I've been working with an ocaml project, that

(a) pulls the current git tag (if it's of the form KK.NN.MM) and stores that (and displays for version)
(b) also pulls the current git sha and again stores/displays that

By "store" I mean, that a build-time script creates a source-code file with the "version-numbers" in it.  And by "display" I mean (of course) that some "-version" flag prints this version-number.

Seems like a decent way to go.  Prints out release versions nicely, and when not at a release version, you can fall back to the commit sha.
